crates/ruff_linter/src/rules/ruff/rules/helpers.rs

```diff
@@ -200,3 +201,25 @@ pub(super) fn is_descriptor_class(func: &Expr, semantic: &SemanticModel) -> bool
         })
     })
 }
+
+/// Returns the inferred value of the given annotation binding of a given [`Expr`].
+///
+/// If an annotation [`Expr`] is bound to a type alias, this function will return the inferred
+/// value of the type alias.
+pub(super) fn map_annotation_binding<'a>(expr: &'a Expr, semantic: &'a SemanticModel) -> &'a Expr {
+    let Some(expr_name) = expr.clone().name_expr() else {
+        return expr;
+    };
+
+    let Some(binding) = semantic
+        .resolve_name(&expr_name)
+        .map(|id| semantic.binding(id))
+    else {
+        return expr;
+    };
+
+    match find_binding_value(binding, semantic) {
+        Some(value) => map_annotation_binding(map_subscript(value), semantic),
+        None => expr,
+    }
```

Comment:
Should we use `find_assigned_value` instead? This will only look in the current scope and will work with forward references as well.

https://github.com/astral-sh/ruff/blob/f58a54f043aab836706a22eb3f9b4d450477ba99/crates/ruff_python_semantic/src/analyze/typing.rs#L1137-L1151

Reply:
I tested with `find_assigned_value` but its not fixing the forwarded value. If you import the type it does not resolves it completely.

I tried using multiple method with `ResolvedReferences` and with `resolve_qualified_name` but I didn't had any success on this. 

Is there any function that resolves a `FromImport` to an `Exp` somehow ?